### PR TITLE
Add spec for deleting/disabling users

### PR DESCRIPTION
**Pull request by kiootic**
ref #148 
ref #150

### CODE REVIEW

File: docs/specs/disable-user.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Admin may optionally provide a reason when disabling user. This reason would be
+shown when user attempted to login.
+
+## Disabled user session
+
+When a user is disabled, the user can no longer create new sessions. However,
+there may be existing logged in sessions.
+
+The authentication tokens (i.e. sessions, access token, refresh tokens, etc.)
```
Comment (louischan-oursky) [inline docs/specs/disable-user.md:21]: Do we update all token at once or do we check user disabled status every time we resolve a token?

File: docs/specs/delete-user.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Deleting user
+
+A user can be deleted by admins. A deleted user would have all its associated
+data erased from database (hard-delete), including identities, authenticators,
+password data, etc. 
+
+## Cached data
+
+Some data may be present in caches (Redis), such as sessions, rate limit
```
Comment (louischan-oursky) [inline docs/specs/delete-user.md:15]: We should also remove sessions as part of the deletion process, as we treat sessions as entity as well.

File: docs/specs/disable-user.md (hunk 1 above)
Comment (kiootic) [reply docs/specs/disable-user.md:21]: I intended to check disabled status for every resolver request for accuracy. Consumer may cache the resolver result/use planned JWT access token for performance.

File: docs/specs/delete-user.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+Some internal data may be present in cache (Redis), such as OAuth states,
+MFA device tokens, rate limit counter. These data would remain in the cache
+until its natural expiry.
+
+## Future works
+
+- Web-hooks
+- Reserve deleted login ID
+- Self-service deletion
```
Comment (chpapa) [inline docs/specs/delete-user.md:23]: A common feature needed is "Soft delete for N days and delete", I think it sort of like "Disable for N days and then delete" in our case. Maybe put it in future work so it is considered? (Although it should be relatively trivial)

Comment (chpapa) [reply docs/specs/delete-user.md:23]: BTW, same applies to self-service deleteion too, for a lot of consumer apps the Business would prefer to let the user delete but have a 30 days period to revert its decision. In those case we need to display a better default copywriting of reason for disable users for this case.

**Review by chpapa — APPROVED** [review]
Doesn't see any problem besides want to add a note on future work.

**Pull request merged**